### ISSUE TITLE
Paket pack failing in multi-target projects

### ISSUE THREAD

**Issue by eiriktsarpalis** [fsprojects/Paket#3047]
### Description

It seems like the paket pack command is failing when attempting to bundle packages in multi-target projects. 

### Repro steps

1. Do a Release build of [this project](https://github.com/eiriktsarpalis/Repro/tree/acaa51c4f220fb587782af707e9e874f65a22817).

2. From the command line, call `.paket/paket.exe pack .`

### Expected behavior

Should construct a nuget package with supplied parameters.

### Actual behavior

Fails with the following error:

```
Paket failed with
-> Unable to find Release output path node in file C:\Users\eirik.tsarpalis\Source\Repos\Repro\Library\Library.fsproj for any known platforms
```

### Known workarounds

None known.

**Comment by forki** [comment]
Haha @matthid this is hunting us now.

Am 14.02.2018 17:14 schrieb "Eirik Tsarpalis" <notifications@github.com>:

**Comment by eiriktsarpalis** [comment]
@forki context? I'll see if I can write a fix for this over the weekend.

**Comment by eiriktsarpalis** [comment]
I have been able to make it work in a non multi-target setting.

**Comment by eiriktsarpalis** [comment]
Quite a bit of unnecessary boilerplate though: https://github.com/fsprojects/Argu/blob/35ff826bb0944d06cb2763f79d4a86d48d83b643/src/Argu/paket.template

**Comment by forki** [comment]
Let's just say pack command needs a bit of help. But especially for dotnet
core people used dotnet pack with success, so it wasn't our highest
priority.

Am 14.02.2018 20:06 schrieb "Eirik Tsarpalis" <notifications@github.com>:

**Comment by eiriktsarpalis** [comment]
Makes sense. I had evaluated dotnet pack, but wasn't too impressed by its configurability.

**Comment by matthid** [comment]
What exactly is missing in dotnet pack?

**Comment by eiriktsarpalis** [comment]
Couldn't find a way to pass version numbers or customize contents, e.g. to
include DLLs from other projects.

**Comment by enricosada** [comment]
@eiriktsarpalis you can include any files, in any nupkg directory (renamed too) with something similar to

```
    <Content Include="build/FSharp.NET.Sdk.props">
      <PackagePath>build/FSharp.NET.Sdk.props</PackagePath>
      <Pack>true</Pack>
    </Content>
```

You can also include output of projectreferences with a different xml thing.

version number like what? `/p:Version` and `/p:PackageVersion` can customize some.

**Comment by eiriktsarpalis** [comment]
@enricosada thanks for the info. I'd still be looking for a replacement to the `CURRENTVERSION` and `LOCKEDVERSION` macros that paket offers (c.f. https://fsprojects.github.io/Paket/template-files.html#Dependencies)

**Comment by enricosada** [comment]
`CURRENTVERSION` and `LOCKEDVERSION` yes are missing.

Root issue is, where are config/options for `paket` pack? 
Doing `dotnet pack` works with paket who manage the dependencies list (that's the integration).
Metadata are now already now handled by sdk (`Authors` etc), deps too `paket.references`).
But where can users write `CURRENTVERSION` and `LOCKEDVERSION`?

Atm is `paket.template` considered when doing `dotnet pack`? ihmo it should at least partially override set properties

Two options i think:

1) inside fsproj

```xml
<PaketDep Include="Argu" PackVersion="LOCKEDVERSION" />
```

this is easy to implement:

- on paket pack target (where there props can be read), write an `obj\paket.pack.{framework}.csv` for that info as first thing
- continuing the paket target, it call paket to generate the deps (like now), but now consider also that file if exists

2) outside in `paket.template`

make `paket.template` like a partial file, so can override any property (if set). 
Implemented like 1, that file is considered when resolving what paket deps to pass to Pack task. And addiotional file can be used to override metadata

Implementation wise both are same complexity, but maybe the second is easier as first step and easier for current users

**Comment by jam40jeff** [comment]
@forki This should be fixed along with #3494.  (At least the original issue should be fixed, I believe the later discussion is covered by #2883.)
